File: 0009-palindrome-number/0009-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        x = str(x)
        left = 0
        right = len(x) - 1

        flag = True

        while left < right:
            if x[left] != x[right]:
                flag = False
                break
            left += 1
            right -= 1
            
        
        return flag

        

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0){
            return false;
        }

        string digit = std::to_string(x);
        int left = 0;
        int right = digit.length() - 1;
        while(left < right){
            if(digit[left] != digit[right]){
                return false;
            }

            left++;
            right--;
        }
        return true;
        
    }
};